File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/atendimento/dto/AtendimentoDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.atendimento.dto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class AtendimentoDTO implements DTO {

    private UUID id;
    private Long numero;
    private LocalDateTime dataInicio;
    private LocalDateTime dataFim;

    private UUID setorId;
    private String setorNome;

    private UUID pacienteId;
    private String pacienteNome;

    private UUID responsavelId;
    private String responsavelNome;

    private UUID convenioId;
    private String convenioNome;

    private UUID convenioCategoriaId;
    private String convenioCategoriaNome;

    private UUID profissionalAtendimentoId;
    private String profissionalAtendimentoNome;

    private UUID profissionalResponsavelId;
    private String profissionalResponsavelNome;

    private List<AtendimentoProcedimentoDTO> procedimentos;

    private String observacoes;

    private UUID lancamentoFinanceiroId;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: src/frontend/src/app/components/atendimento/atendimento/model/atendimento-dto.ts
import { Exclude, Expose } from 'class-transformer';
import { AtendimentoProcedimentoDTO } from './atendimento-procedimento-dto';

@Exclude()
export class AtendimentoDTO {
  @Expose() id?: string;
  @Expose() numero?: number;
  @Expose() dataInicio?: string;
  @Expose() dataFim?: string;

  @Expose() setorId?: string;
  @Expose() setorNome?: string;

  @Expose() pacienteId?: string;
  @Expose() pacienteNome?: string;

  @Expose() responsavelId?: string;
  @Expose() responsavelNome?: string;

  @Expose() convenioId?: string;
  @Expose() convenioNome?: string;

  @Expose() convenioCategoriaId?: string;
  @Expose() convenioCategoriaNome?: string;

  @Expose() profissionalAtendimentoId?: string;
  @Expose() profissionalAtendimentoNome?: string;

  @Expose() profissionalResponsavelId?: string;
  @Expose() profissionalResponsavelNome?: string;

  @Expose() procedimentos?: AtendimentoProcedimentoDTO[];

  @Expose() observacoes?: string;

  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}
